### PR TITLE
Add Structured Text Insights Flow with Dynamic Extension Demo

### PR DESCRIPTION
**Pull request by shivchander**
## Summary

This PR adds a comprehensive **Structured Text Insights Flow** to SDG Hub with an innovative demonstration of dynamic flow extension capabilities.

### 🎯 Core Features Added

- **📝 JSONStructureBlock**: New transform block for combining multiple analysis results into structured JSON output
- **🔄 Text Analysis Flow**: Complete pipeline extracting summaries, keywords, entities, and sentiment from text
- **📊 Bloomberg Financial News Integration**: Real-world example using 447k financial news articles (2006-2013)
- **🚀 Dynamic Flow Extension Demo**: Shows how to extend flows at runtime without modifying core files

### 🔧 Dynamic Extension Demonstration

The key innovation is demonstrating **runtime flow modification** using existing blocks:

- **PromptBuilderBlock** - Creates stock ticker extraction prompts
- **LLMChatBlock** - Processes extraction via LLM  
- **TextParserBlock** - Parses LLM output to clean results
- **JSONStructureBlock** - Combines all insights with new fields

### 📈 Enhanced Output Example

**Basic insights:**
```json
{
  "summary": "Brief summary...",
  "keywords": "keyword1, keyword2...",
  "entities": "Entity1, Entity2...",
  "sentiment": "positive"
}
```

**Enhanced with stock tickers:**
```json
{
  "summary": "Brief summary...", 
  "keywords": "keyword1, keyword2...",
  "entities": "Entity1, Entity2...",
  "sentiment": "positive",
  "stock_tickers": "AAPL, MSFT, GOOGL"
}
```

### 📁 Files Added

- `src/sdg_hub/core/blocks/transform/json_structure_block.py` - New JSONStructureBlock
- `src/sdg_hub/flows/text_analysis/structured_insights/` - Complete text analysis flow
- `examples/AIIT/text_analysis/structured_insights_demo.ipynb` - Comprehensive demo notebook
- `examples/AIIT/text_analysis/README.md` - Documentation and usage guide

### 🧪 Test Coverage

- 14 comprehensive test cases for JSONStructureBlock (100% passing)
- End-to-end flow validation with Bloomberg dataset
- Error handling and edge case coverage

## Test plan

- [x] All existing tests pass
- [x] New JSONStructureBlock tests pass (14/14)
- [x] Flow discovery and loading works correctly
- [x] Bloomberg dataset integration successful
- [x] Stock ticker extraction demo functional
- [x] Enhanced JSON output validates correctly
- [x] Documentation complete and accurate

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Structured Text Insights flow that extracts summary, keywords, entities, and sentiment and emits a single JSON field.
  * New prompt for extracting stock tickers and dedicated prompts for summary, keywords, entities, and sentiment.
  * New configurable transform block to assemble multiple fields into one JSON output column.

* **Documentation**
  * New text-analysis README with quick start, model options, scaling, customization, and runtime extension guidance.

* **Tests**
  * Extensive tests validating JSON assembly across types, edge cases, serialization, and performance.

### CODE REVIEW

File: src/sdg_hub/core/blocks/transform/json_structure_block.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                json_field: col_name for col_name, json_field in self.input_cols.items()
+            }
+
+        # If input_cols is a list, use column names as JSON field names
+        if isinstance(self.input_cols, list):
+            return {col: col for col in self.input_cols}
+
+        raise ValueError("Unable to determine field mapping from input_cols")
+
+    def generate(self, samples: Dataset, **kwargs: Any) -> Dataset:
```
Comment (mihirathale98) [inline src/sdg_hub/core/blocks/transform/json_structure_block.py:93]: Would we like to add override args support/handling for this block as well at runtime?

File: examples/AIIT/text_analysis/structured_insights_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,4307 @@
+{
```
Comment (eshwarprasadS) [inline examples/AIIT/text_analysis/structured_insights_demo.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    first_result = results[random.randint(0, len(results) - 1)]
</pre><p> </p><p>Just a nit: this is technically a random result out of the results, not the first result, maybe worth renaming this for clarity. No functional blocker though.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Red-Hat-AI-Innovation-Team/sdg_hub/pull/409/discussion/#comment-2345214371'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

Comment (eshwarprasadS) [inline examples/AIIT/text_analysis/structured_insights_demo.ipynb:1]: <p>I ran the notebook end to end as well, and I also ended up with 45 generated rows for an input of 50 rows. Losing some rows to parsing is possible, but for a mostly structured straightforward inference like this one, I'd expect near 100% parse rate, similar to the annotation one for eg. Just wondering why this is not the case.</p><p> </p><p>If it turns out to be complex parsing, maybe we can note that there will be few rows missing from input -&gt; output due to variance? Just a thought.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Red-Hat-AI-Innovation-Team/sdg_hub/pull/409/discussion/#comment-2345214376'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

Comment (eshwarprasadS) [inline examples/AIIT/text_analysis/structured_insights_demo.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    first_result2 = results2[random.randint(0, len(results2) - 1)]
</pre><p>Same nit comment as before</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Red-Hat-AI-Innovation-Team/sdg_hub/pull/409/discussion/#comment-2345214381'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='23'/>

Comment (mihirathale98) [reply examples/AIIT/text_analysis/structured_insights_demo.ipynb:1]: For me it is failing quite a few times in the Text Parser block while trying to extract stock symbols. 

Suggestion - 

We can improve the prompt as most times, model is returning "None" without the tags.

Comment (eshwarprasadS) [reply examples/AIIT/text_analysis/structured_insights_demo.ipynb:1]: Fixed in a new commit
